internal/impl/aws/input_sqs.go

```diff
@@ -440,48 +541,79 @@ func (a *awsSQSReader) deleteMessages(ctx context.Context, msgs ...sqsMessageHan
 			return err
```

Comment:
Just wondering, if there's some transitive error and `DeleteMessageBatch` fails, we return an error here. However, further below, we only log an error when `fail.Message != nil`. Can that lead to failed messages being retried over and over?